packages/artifact/src/internal/artifact-client.ts

```diff
@@ -114,10 +122,27 @@ export class DefaultArtifactClient implements ArtifactClient {
 
       // Update the size of the artifact to indicate we are done uploading
       // The uncompressed size is used for display when downloading a zip of the artifact from the UI
+      core.info(
+        `File upload process has finished. Finalizing the artifact upload`
+      )
       await uploadHttpClient.patchArtifactSize(uploadResult.totalSize, name)
 
+      if (uploadResult.failedItems.length > 0) {
+        core.info(
```

Comment:
Could this be a warning or is this scenario expected to happen?

Reply:
So there is a `continueOnError` option that we have available: https://github.com/actions/toolkit/tree/main/packages/artifact#available-options

It's not really used and we default to `false` for that so this is mainly for action authors that are doing some weird stuff. If that is `true` then it would error out earlier. The existing library also returns an array of `failedItems`: https://github.com/actions/toolkit/tree/main/packages/artifact#upload-result so it's basically up to authors to do whatever they want with that failed upload result list.

I lean on not creating a warning because it shows up as an annotation and in the past there was pushback about warning annotations we made if there were no files found with the input path